examples/private_connection/main.tf

```diff
@@ -0,0 +1,189 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+provider "google" {
+  version = "~> 3.7.0"
+  project = var.project_id
+}
+
+locals {
+  service_list = [
+    "compute.googleapis.com",
+    "cloudresourcemanager.googleapis.com",
+    "serviceusage.googleapis.com",
+    "servicenetworking.googleapis.com",
+    "dns.googleapis.com"
+  ]
+}
+
+resource "google_project_service" "main" {
+  count              = length(local.service_list)
+  project            = var.project_id
+  service            = local.service_list[count.index]
+  disable_on_destroy = "false"
+}
+
+resource "google_compute_network" "network" {
+  name                    = "forseti-network"
+  auto_create_subnetworks = false
+}
+
+resource "google_compute_subnetwork" "subnet" {
+  name                     = "forseti-subnet"
+  region                   = var.region
+  ip_cidr_range            = "10.127.0.0/20"
+  network                  = google_compute_network.network.self_link
+  private_ip_google_access = true
+  log_config {
+    aggregation_interval = "INTERVAL_10_MIN"
+    flow_sampling        = 1
+    metadata             = "INCLUDE_ALL_METADATA"
+  }
+}
+
+# Router to control egress connectivity
```

Comment:
From the test failure
```
terraform_validate ./examples/private_connection 

Error: Missing resource instance key

  on main.tf line 68, in resource "google_compute_router_nat" "forseti-nat":
  68:   router = google_compute_router.forseti-router.name

Because google_compute_router.forseti-router has "count" set, its attributes
must be accessed on specific instances.

For example, to correlate with indices of a referring resource, use:
    google_compute_router.forseti-router[count.index]


Error: Missing resource instance key

  on main.tf line 69, in resource "google_compute_router_nat" "forseti-nat":
  69:   region = google_compute_router.forseti-router.region

Because google_compute_router.forseti-router has "count" set, its attributes
must be accessed on specific instances.

For example, to correlate with indices of a referring resource, use:
    google_compute_router.forseti-router[count.index]
```

Reply:
fixed